src/pow_fpga_accel.c

```diff
@@ -0,0 +1,158 @@
+/*MIT License
+
+Copyright (c) 2018 Ievgen Korokyi
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.*/
+
+#include "pow_fpga_accel.h"
+
+
+#define NONCE_LEN 81
+#define TRANSACTION_LEN 2673
+#define NONCE_OFFSET 2646
+#define HPS_TO_FPGA_BASE 0xC0000000
+#define HPS_TO_FPGA_SPAN 0x0020000
+#define HASH_CNT_REG_OFFSET 4
+#define TICK_CNT_LOW_REG_OFFSET 5
+#define TICK_CNT_HI_REG_OFFSET 6
+#define MWM_MASK_REG_OFFSET 3
+#define CPOW_BASE 0
+
+#define HINTS                                                           \
+    "### CPOW Hardware Accelerated ###\nUsage:\n\t./curl_pow_hard MWM " \
+    "TRYTES(length: %d) \n"
+
+static int8_t otrytes[TRANSACTION_LEN];
+
+int pow_fpga_accel_init()
+{
+    ctrl_fd = 0;
+    in_fd = 0;
+    out_fd = 0;
+    devmem_fd = 0;
+    fpga_regs_map = 0;
+    cpow_map = 0;
+
+    ctrl_fd = fopen("/dev/cpow-ctrl", "r+");
```

Comment:
I do not known what you mean.

Reply:
`/dev/cpow-ctrl` is a device file created by the kernel module, which communicates to FPGA. It would be better if we define several macros to specify these device files in advance.